From 1f6536f2fdbf7fff9fa476382ac10fa36119f321 Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hias@horus.com>
Date: Thu, 25 Feb 2016 14:18:03 +0100
Subject: cirrus: silence probe deferral messages from register card

The reason for the probe deferral (eg DAI not registered) is already
logged, so there's no need for further dmesg spam.
---
 sound/soc/bcm/rpi-cirrus-sound-pi.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/sound/soc/bcm/rpi-cirrus-sound-pi.c b/sound/soc/bcm/rpi-cirrus-sound-pi.c
index eeb41af..3abfff1 100644
--- a/sound/soc/bcm/rpi-cirrus-sound-pi.c
+++ b/sound/soc/bcm/rpi-cirrus-sound-pi.c
@@ -590,7 +590,11 @@ static int snd_rpi_wsp_probe(struct platform_device *pdev)
 	snd_rpi_wsp.dev = &pdev->dev;
 	ret = snd_soc_register_card(&snd_rpi_wsp);
 	if (ret) {
-		dev_err(&pdev->dev, "Failed to register card: %d\n", ret);
+		if (ret == -EPROBE_DEFER)
+			dev_dbg(&pdev->dev, "register card requested probe deferral\n");
+		else
+			dev_err(&pdev->dev, "Failed to register card: %d\n", ret);
+
 		kfree(wm5102);
 	}
 
-- 
2.9.0

